### PR TITLE
Api ssl

### PR DESCRIPTION
**Pull request by witjoh**
makes it possible to connect using ssl.

### CODE REVIEW

File: lib/puppet/provider/graylog_api/graylog_api.rb [hunk 1]
```diff
@@ -21,21 +21,38 @@ def self.prefetch(resources)
...
+
+    # Settings needed before connection to the server.  We set them on the global
+    # level, so we only need to define them once here.
+    if tls
+      HTTParty::Basement.default_options.update(verify: verify_tls)
+      # we configure this anyway, even we don't want to verify the certificate
+      HTTParty::Basement.default_options.update(ssl_ca_file: ssl_ca_file)
+    end
+    wait_for_api(port, server)
+    wait_for_api(port, server)
```
Comment (philomory) [inline lib/puppet/provider/graylog_api/graylog_api.rb:43]: Is there a reason you call `wait_for_api` twice in rapid succession, here?

File: lib/puppet/provider/graylog_api/graylog_api.rb [hunk 2]
```diff
@@ -21,21 +21,38 @@ def self.prefetch(resources)
...
+    Puppet::Provider::GraylogAPI.api_server = server
+    Puppet::Provider::GraylogAPI.verify_tls = verify_tls
+    Puppet::Provider::GraylogAPI.ssl_ca_file = ssl_ca_file
+
+    # Settings needed before connection to the server.  We set them on the global
+    # level, so we only need to define them once here.
+    if tls
+      HTTParty::Basement.default_options.update(verify: verify_tls)
+      # we configure this anyway, even we don't want to verify the certificate
+      HTTParty::Basement.default_options.update(ssl_ca_file: ssl_ca_file)
```
Comment (philomory) [inline lib/puppet/provider/graylog_api/graylog_api.rb:40]: I'm a little conflicted on this. On the one hand, it's much more concise and convenient than any alternative I can think of; on the other hand, in the (admittedly unlikely) event that there's another Puppet provider used in the same catalog that also uses HTTParty, won't this interfere with the operation of that provider?

**Review by philomory — COMMENTED** [review]
Hey, thanks for submitting this! I've got a few questions/comments, but other than those this looks pretty good to me.

File: lib/puppet/provider/graylog_api/graylog_api.rb (hunk 1 above)
Comment (witjoh) [reply lib/puppet/provider/graylog_api/graylog_api.rb:43]: No reason at all.  Think I yanked one line too much .. will remove one

File: lib/puppet/provider/graylog_api/graylog_api.rb (hunk 2 above)
Comment (witjoh) [reply lib/puppet/provider/graylog_api/graylog_api.rb:40]: Makes it possible to only define these  setting once.  I suspect if we choose not to do it this way, we need to add all the settings to the other providers.

Comment (philomory) [reply lib/puppet/provider/graylog_api/graylog_api.rb:40]: All of the other providers in the module inherit from this one, so we should be able to define our options centrally and then take advantage of them throughout the module, without setting process-wide defaults that might interfere with providers in other modules. That said, I'm happy to merge this now and then refactor it myself.

**Review by philomory — APPROVED** [review]

**Comment by philomory** [comment]
@witjoh Would you care to check out the branch `chore/refactor-tls-support` to see if it works for you?

**Comment by philomory** [comment]
@witjoh Try now? The method is named simply `ssl_ca_file`, not `api_ssl_ca_file`; I've fixed it in the `chore/refactor-tls-support` branch.

**Comment by philomory** [comment]
Actually, I went ahead and removed the method call, because after the refactoring the variable in which the value of that method call was being stored, is never used after it's assigned anyway. I left the method definition in place for consistency with the other attributes, though.

**Comment by witjoh** [comment]
Just tested the  chore/refactor-tls-support branch.  Api calls are now working.  Thanks ..

**Comment by philomory** [comment]
@witjoh Thanks for the confirmation. Version 0.5.0 has now been released to Puppet Forge

**Pull request merged**